### ISSUE TITLE
files .h264 are not purged

### ISSUE THREAD

**Issue by shinji2009** [silvanmelchior/RPi_Cam_Web_Interface#349]
hi
if .h264 or zip file are stays in media folder after error , it is not purged.
even  got one board with 24gb of .h264 and zip files so theres no place for mp4 files left at all

![2216](https://cloud.githubusercontent.com/assets/6340606/26094184/1b6948c2-3a43-11e7-9d4c-ce168789fd6c.GIF)

these 3 zips is because i was trying to download a big mp4 as zip

they can't be seen in interface and everything works like fine, it is only can be discovered if delete all files and theres still no free space.

maybe purge script should first  purge all files in media folders except mp4 and jpg every time it start, and only after that purge old mp4 and jpgs. 

raspbian jessie lite. rpi3b

**Comment by roberttidey** [comment]
Normally the .h264 files get deleted after boxing unless one turns boxing off. 

I don't seem to get any left over but I can see it would gradually become a problem if errors let these build up. The reason for not purging just all .h264 files is that some users turn boxing off as they want to box offline.

I will look at a strategy that does .h264 clean up for the normal case.

**Comment by shinji2009** [comment]
thank you. and zip files too, please. and maybe just any file type that are not needed in this folder

**Comment by roberttidey** [comment]
Couple of purge improvements in the latest version. Note I have to be a bit cautious here as some people store other stuff in the folder from macros etc.

1) .h264 files are now separately checked for when purging or deleting any video related file

2) zip files older than the video purge hours time limit are removed.

**Comment by shinji2009** [comment]
thank you. this  improvements will be implemented soon?

**Comment by roberttidey** [comment]
They are in the current version

**Comment by shinji2009** [comment]
so... h264 file purged only if corresponding mp4 file is present? if theres no mp4 file due to error, h264 file stays forever?

![2230](https://cloud.githubusercontent.com/assets/6340606/26412521/3c96105c-40d3-11e7-8fa2-2fe00569bf32.GIF)

**Comment by roberttidey** [comment]
No. It starts from the thumbnail and deletes all files associated with that including mp4.h264 left over in error.

It won't retrospetively delete old .h264 files left over before the new logic. You have to those manually once. It should clear out from then on.

**Comment by shinji2009** [comment]
it looks like these h264 errors appears only when video setting is 1920*1080. i set to 1280*720 a couple days ago and no busy files since then. also a lot of frames are dropped in a row sometimes. should i create a new issue about it?

**Comment by roberttidey** [comment]
Yes. Create a new issue, but see if you can highlight anything about your specific set up as I haven't seen any specific problems with the higher resolution.

In particular:

Are you storing to the local SD card or some other external storage?

Is the GPU RAM set to at least 128MB?  ( vcgencmd get_mem gpu )

**Comment by shinji2009** [comment]
ok this one should be closed then as i understand.
i'll create a new issue and everything about it will be there.
thank you
